packages/devtools/src/graphQlDirectives/ceramicStringType.ts

```diff
@@ -0,0 +1,44 @@
+import { GraphQLScalarType, GraphQLString, ValueNode } from 'graphql'
+
+export class CeramicStringType extends GraphQLScalarType<string> {
+  constructor (
+    fieldName: string,
+    ceramicTypeName: string,
+    args: Record<string, any>) {
+    super({
+      name: ceramicTypeName,
+      serialize (value: unknown) {
+        const serialized = GraphQLString.serialize(value)
+        validateCeramicString(fieldName, args, serialized)
+        return serialized
+      },
+      parseValue (value: unknown) {
+        const parsed = GraphQLString.serialize(value)
+        validateCeramicString(fieldName, args, parsed)
+        return parsed
+      },
+      parseLiteral (ast: ValueNode) {
+        const value = GraphQLString.parseLiteral(ast)
+        validateCeramicString(fieldName, args, value)
+        return value
+      }
+    })
+  }
+}
+
+function validateCeramicString(
```

Comment:
This seem to perform runtime validation, no?
If so, I don't think we need it at this layer, we'll have validation with JSON schema anyways. What matters most is that we translate the GraphQL validation rules set in directives to their JSON schema equivalents.